lib/danger/ci_source/codeship.rb

```diff
@@ -0,0 +1,44 @@
+# https://semaphoreci.com/docs/available-environment-variables.html
+require "danger/request_sources/github/github"
+
+module Danger
+  # ### CI Setup
+  #
+  # In Codeship, go to your "Project Settings", then add `bundle exec danger` as a test step inside
+  # one of your pipelines.
+  #
+  # ### Token Setup
+  #
+  # Add your `DANGER_GITHUB_API_TOKEN` to "Environment" section in "Project Settings".
+  #
+  class Codeship < CI
+    def self.validates_as_ci?(env)
+      env["CI_NAME"] == "codeship"
+    end
+
+    def self.validates_as_pr?(env)
+      return false unless env["CI_BRANCH"] && !env["CI_BRANCH"].empty?
+
+      !pr_from_env(env, nil).nil?
+    end
+
+    def self.owner_for_github(env)
+      env["CI_REPO_NAME"].split("/").first
+    end
+
+    # this is fairly hacky, see https://github.com/danger/danger/pull/892#issuecomment-329030616 for why
+    def self.pr_from_env(env, ci_source)
+      Danger::RequestSources::GitHub.new(nil, env).get_pr_from_branch(env["CI_REPO_NAME"], env["CI_BRANCH"], owner_for_github(env))
```

Comment:
do we need to pass down the `ci_source` to `Danger::RequestSources::GitHub`? 🤔 
do we need `ci_source`? 😂

Reply:
`ci_source` gets used by a bunch of stuff eg [here](https://github.com/danger/danger/blob/master/lib/danger/request_sources/github/github.rb#L74) and [here](https://github.com/danger/danger/blob/master/lib/danger/request_sources/github/github.rb#L80) - i wasn't keen to take on a full refactor.

Reply:
it's just not required in my particular case - i just need a github client. if you prefer, i can try and find a neater way of writing that.

Reply:
wait, i misread your comment... fixed in https://github.com/danger/danger/pull/892/commits/a49fc306196afe79f69b77fd03eb4f3052cf040d